### DISCUSSION TITLE
reproducable crash of epoptes gui

### DISCUSSION THREAD

**Discussion by mherweg** [epoptes/epoptes#26] [Q&A]
I'm testing version 0.5.10 with 1 fat client.
What I'm especially interested in are the cases where the clients are already running before the epoptes server boots.

the gui crashes when the client shutsdown with this error:
```

Disconnect from 192.168.1.113:32897
Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/twisted/protocols/amp.py", line 889, in _commandReceived
    deferred = self.dispatchCommand(box)
  File "/usr/lib/python2.7/dist-packages/twisted/protocols/amp.py", line 946, in dispatchCommand
    return maybeDeferred(responder, box)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/defer.py", line 139, in maybeDeferred
    result = f(*args, **kw)
  File "/usr/lib/python2.7/dist-packages/twisted/protocols/amp.py", line 1035, in doit
    return maybeDeferred(aCallable, **kw).addCallback(
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/defer.py", line 139, in maybeDeferred
    result = f(*args, **kw)
  File "/usr/lib/python2.7/dist-packages/epoptes/daemon/uiconnection.py", line 56, in clientDisconnected
    self.client.amp_clientDisconnected(handle)
  File "/usr/lib/python2.7/dist-packages/epoptes/ui/gui.py", line 597, in amp_clientDisconnected
    shutdownNotify(client.get_name())
  File "/usr/lib/python2.7/dist-packages/epoptes/ui/notifications.py", line 82, in shutdownNotify
    notify(_("Shut down:"), "%s" %(host), os.path.abspath("images/shutdown.svg"))
  File "/usr/lib/python2.7/dist-packages/epoptes/ui/notifications.py", line 73, in cached_notify
    n.show()
glib.GError: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Notifications was not provided by any .service files

```

**Comment by alkisg** [comment]
It sounds like you don't have any notification daemons installed.
Which distro/version/desktop environment is that?
What's the output of the following command?
  `dpkg -l '*notif*' | grep ^ii`

**Comment by mherweg** [comment]
`
ii  gir1.2-notify-0.7                                           0.7.6-1ubuntu3                                      amd64        sends desktop notifications to a notification daemon (Introspection files)
ii  libknotifyconfig4                                           4:4.14.2-0ubuntu1~ubuntu14.04~ppa1                  amd64        library for configuring KDE Notifications
ii  libnotify-bin                                               0.7.6-1ubuntu3                                      amd64        sends desktop notifications to a notification daemon (Utilities)
ii  libnotify0.4-cil                                            0.4.0~r3032-7                                       all          CLI library for desktop notifications
ii  libnotify4:amd64                                            0.7.6-1ubuntu3                                      amd64        sends desktop notifications to a notification daemon
ii  libstartup-notification0:amd64                              0.12-3ubuntu1                                       amd64        library for program launch feedback (shared library)
ii  notification-daemon                                         0.7.6-1                                             amd64        daemon for displaying passive pop-up notifications
ii  pidgin-libnotify                                            0.14-9ubuntu2                                       amd64        display notification bubbles in pidgin
ii  python-notify                                               0.1.1-3ubuntu2                                      amd64        Python bindings for libnotify
ii  python-pyinotify                                            0.9.4-1build1                                       all          simple Linux inotify Python bindings
ii  update-notifier-common                                      0.154.1ubuntu1                                      all          Files shared between update-notifier and other packages
ii  vlc-plugin-notify                                           2.1.6-0ubuntu14.04.2                                amd64        LibNotify plugin for VLC

`

**Comment by alkisg** [comment]
You have a PPA version there that probably breaks notifications, please try to reproduce the issue with a clean Ubuntu Trusty installation.
